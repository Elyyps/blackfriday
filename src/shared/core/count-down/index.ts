export * from "./components/count-down.component";
