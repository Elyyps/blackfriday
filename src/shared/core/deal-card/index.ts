export * from "./deal-card.component";
