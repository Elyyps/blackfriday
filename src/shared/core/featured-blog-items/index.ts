export * from "./featured-blog-items.component";
