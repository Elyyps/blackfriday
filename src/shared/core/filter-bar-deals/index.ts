export * from "./filter-bar-deals.component";
export * from "./container";
