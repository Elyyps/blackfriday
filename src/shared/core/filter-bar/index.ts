export * from "./filter-bar.component";
export * from "./container";
