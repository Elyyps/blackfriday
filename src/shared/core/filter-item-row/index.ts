export * from "./filter-item-row.component";
