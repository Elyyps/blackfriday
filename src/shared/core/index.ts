export * from "./button";
export * from "./image";
export * from "./usp";
export * from "./icon";
