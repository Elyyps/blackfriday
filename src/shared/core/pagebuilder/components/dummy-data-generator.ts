import { PageModel } from "@app/api/pagebuilder/page.model";
import { dummyFeaturedShopsData } from "@app/api/modules/featured-shops/generate-dummy-data";

export const generatePageModel = (id: string, addWordPressModules: boolean): PageModel => ({
  metaData: {
    description: "",
    content: "",
    title: ""
  },
  id,
  route: "",
  wordPressPostModules: addWordPressModules ? [dummyFeaturedShopsData()] : []
});
