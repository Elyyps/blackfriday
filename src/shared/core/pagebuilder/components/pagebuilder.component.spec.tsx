import React from "react";
import { shallow } from "enzyme";

import { PagebuilderComponent, IPagebuilderComponentProps } from "@app/core/pagebuilder";
import { getMockRouterProps } from "@app/util/get-mock-router-props";
import { generatePageModel } from "./dummy-data-generator";
jest.mock("@app/api", () => "");

describe("[PagebuilderComponent]", () => {
  it("should render without modules without crashing", () => {
    const routerProps = getMockRouterProps<IPagebuilderComponentProps>();
    const page = generatePageModel("1", false);

    const renderedComponent = shallow(
      <PagebuilderComponent
        history={routerProps.history}
        location={routerProps.location}
        staticContext={routerProps.staticContext}
        match={routerProps.match}
        currentPage={page}
        getPage={jest.fn()}
        screenSize={undefined}
        setScreenSize={jest.fn()}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it("should render with 2 modules without crashing", () => {
    const routerProps = getMockRouterProps<IPagebuilderComponentProps>();
    const page = generatePageModel("1", true);

    const renderedComponent = shallow(
      <PagebuilderComponent
        history={routerProps.history}
        location={routerProps.location}
        staticContext={routerProps.staticContext}
        match={routerProps.match}
        currentPage={page}
        getPage={jest.fn()}
        screenSize={undefined}
        setScreenSize={jest.fn()}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });
});
