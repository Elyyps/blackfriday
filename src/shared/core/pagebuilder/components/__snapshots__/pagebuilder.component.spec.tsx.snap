// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PagebuilderComponent] should render with 2 modules without crashing 1`] = `
<Fragment>
  <HelmetComponent
    content=""
    description=""
    title=""
  />
  <ModuleComponent
    isMobile={false}
    key="0"
    wordPressModule={
      Object {
        "bottomMargin": "0px",
        "id": "1",
        "name": "FeaturedStoresModule",
        "seeMoreCard": Object {
          "icon": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/store.svg",
          "link": Object {
            "title": "Alle winkels",
            "url": "/",
          },
          "title": "Wij hebben meer winkels in ons overzicht",
        },
        "shops": Array [
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.8,
            },
            "title": "Bekbeddiscounter.nl",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "ACTIE START OVER 5 DAGEN!",
              "value": 0,
            },
            "title": "Swiss Sense",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 1,
            },
            "title": "Beter Bed",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.2,
            },
            "title": "Beter Bed",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.2,
            },
            "title": "Beter Bed",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.2,
            },
            "title": "Beter Bed",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.2,
            },
            "title": "Beter Bed",
          },
          Object {
            "button": Object {
              "title": "Naar deals",
              "url": "/deals-overview",
            },
            "content": "Een seletie van bizarre vele aanbiedingen.",
            "picture": "https://d2ddoduugvun08.cloudfront.net/items/2c3217062N3d2L320w0U/Image%202019-06-27%20at%203.17.28%20PM.png?X-CloudApp-Visitor-Id=3342233&v=03808fb0",
            "seeMore": Object {
              "title": "Meer info",
              "url": "/stores-single",
            },
            "timeLeftBar": Object {
              "text": "Nu geldig",
              "value": 0.2,
            },
            "title": "Beter Bed",
          },
        ],
        "title": "Vergelijkbare winkels",
        "topMargin": "0px",
      }
    }
  />
</Fragment>
`;

exports[`[PagebuilderComponent] should render without modules without crashing 1`] = `
<Fragment>
  <HelmetComponent
    content=""
    description=""
    title=""
  />
</Fragment>
`;
